### ISSUE TITLE
Update to Scala 2.13.0

### ISSUE THREAD

**Issue by iamrecursion** [luna/luna#926]
### Summary
Currently the project builds against Scala 2.12.8, but we would like for it to build against 2.13.0 due to the improvements that release brings to the collections API, compiler performance, and the performance of generated code. 

Doing this is currently blocked by some dependencies not yet providing versions built for 2.13.0. This should happen soon, so this task is on hold for a couple of days while the dependency builds catch up. 

### Specification
- Update the project to use the 2.13.0 compiler version.
- Ensure that all dependencies build with this version

### Acceptance Criteria & Test Cases
- [x] Enso builds using Scala 2.13.0.
